sklearn/ensemble/_weight_boosting.py

```diff
@@ -487,19 +480,14 @@ class AdaBoostClassifier(
     refer to :ref:`sphx_glr_auto_examples_ensemble_plot_adaboost_twoclass.py`.
     """
 
-    # TODO(1.8): remove "algorithm" entry
-    _parameter_constraints: dict = {
-        **BaseWeightBoosting._parameter_constraints,
-        "algorithm": [StrOptions({"SAMME"}), Hidden(StrOptions({"deprecated"}))],
-    }
+    _parameter_constraints: dict = {**BaseWeightBoosting._parameter_constraints}
```

Comment:
Then we can simply remove this, right?

Reply:
Ah yes indeed, I did not pay attention. Since all the parameters are fed to the parent class, they will get validated in the parent init, right ?